gordo_components/server/server.py

```diff
@@ -123,10 +131,56 @@ def base_healthcheck():
     return app
 
 
-def run_server(host: str = "0.0.0.0", port: int = 5555, debug: bool = False):
-    app = build_app()
-    app.run(host, port, debug=debug, threaded=False)
+def run_server(
+    host: str, port: int, workers: int, worker_connections: int, log_level: str
+):
+    """
+    Run application with Gunicorn server using Gevent Async workers
+
+    Parameters
+    ----------
+    host: str
+        The host to run the server on.
+    port: int
+        The port to run the server on.
+    workers: int
+        The number of worker processes for handling requests.
+    worker_connections: int
+        The maximum number of simultaneous clients per worker process.
+    log_level: str
+        The log level for the `gunicorn` webserver. Valid log level names can be found
+        in the [gunicorn documentation](http://docs.gunicorn.org/en/stable/settings.html#loglevel).
+    """
 
+    cmd = [
+        "gunicorn",
+        "--bind",
+        f"{host}:{port}",
+        "--log-level",
+        log_level,
+        "--worker-class",
+        "gevent",
+        "--worker-tmp-dir",
+        "/dev/shm",
+        "--workers",
+        str(workers),
+        "--worker-connections",
+        str(worker_connections),
+        "gordo_components.server.server:app",
+    ]
+
+    try:
+        subprocess.check_call(cmd)
+    except subprocess.CalledProcessError as e:
+        logger.exception(e)
+        raise
+    except OSError as e:
+        logger.exception(f"{e}: Check that Gunicorn is installed.")
+        raise
+
+
+app = build_app()
```

Comment:
Should this not be in the `__main__` block?

Reply:
This was for referencing the `app` in the gunicorn call:
`"gordo_components.server.server:app",`

I'd initially created a new module for doing, but then decided this was perhaps simpler. Maybe that is better?

**production.py**
```
from gordo_components.server.server import build_app

app = build_app()
```
**server.py**
```
def run_server():
   cmd = ["gunicorn",  "gordo_components.server.production:app"]
```

Reply:
Does that seem OK to you, or should I do it a different way you think?